checkPropTypes.js

```diff
@@ -28,7 +28,7 @@ if (process.env.NODE_ENV !== 'production') {
 function checkPropTypes(typeSpecs, values, location, componentName, getStack) {
   if (process.env.NODE_ENV !== 'production') {
     for (var typeSpecName in typeSpecs) {
-      if (typeSpecs.hasOwnProperty(typeSpecName)) {
+      if (Object.prototype.hasOwnProperty.call(typeSpecs, typeSpecName)) {
```

Comment:
The next robustness step is to do `var has = Object.prototypr.hasOwnProperty` at module scope, and then here, do `has.call`.